From 3275b4df3c39f97ee3fe982c9bafd6f3b7ff0dfe Mon Sep 17 00:00:00 2001
From: "David S. Miller" <davem@davemloft.net>
Date: Mon, 17 Sep 2018 18:43:42 -0700
Subject: [PATCH 120/329] Revert "kcm: remove any offset before parsing
 messages"

This reverts commit 072222b488bc55cce92ff246bdc10115fd57d3ab.

I just read that this causes regressions.

Signed-off-by: David S. Miller <davem@davemloft.net>
---
 net/kcm/kcmsock.c | 26 +-------------------------
 1 file changed, 1 insertion(+), 25 deletions(-)

diff --git a/net/kcm/kcmsock.c b/net/kcm/kcmsock.c
index 36c438b95955..571d824e4e24 100644
--- a/net/kcm/kcmsock.c
+++ b/net/kcm/kcmsock.c
@@ -381,32 +381,8 @@ static int kcm_parse_func_strparser(struct strparser *strp, struct sk_buff *skb)
 {
 	struct kcm_psock *psock = container_of(strp, struct kcm_psock, strp);
 	struct bpf_prog *prog = psock->bpf_prog;
-	struct sk_buff *clone_skb = NULL;
-	struct strp_msg *stm;
-	int rc;
-
-	stm = strp_msg(skb);
-	if (stm->offset) {
-		skb = clone_skb = skb_clone(skb, GFP_ATOMIC);
-		if (!clone_skb)
-			return -ENOMEM;
-
-		if (!pskb_pull(clone_skb, stm->offset)) {
-			rc = -ENOMEM;
-			goto out;
-		}
-
-		/* reset cloned skb's offset for bpf programs using it */
-		stm = strp_msg(clone_skb);
-		stm->offset = 0;
-	}
-
-	rc = (*prog->bpf_func)(skb, prog->insnsi);
-out:
-	if (clone_skb)
-		kfree_skb(clone_skb);
 
-	return rc;
+	return (*prog->bpf_func)(skb, prog->insnsi);
 }
 
 static int kcm_read_sock_done(struct strparser *strp, int err)
-- 
2.18.0

